pydicom/config.py

```diff
@@ -152,15 +153,93 @@ def DS_decimal(use_Decimal_boolean: bool = True) -> None:
 Default ``False``.
 """
 
+
 enforce_valid_values = False
-"""Raise exceptions if any value is not allowed by DICOM Standard.
+"""Deprecated.
+Use :attr:`Settings.reading_validation_mode` instead.
+"""
 
-e.g. DS strings that are longer than 16 characters; IS strings outside
-the allowed range.
 
-Default ``False``.
+# Constants used to define how data element values shall be validated
+NO_CHECK = 0
+"""If one of the validation modes is set this value, no value validation
+will be performed.
+"""
+
+WARN = 1
+"""If one of the validation modes is set to this value, a warning is issued if
+a value validation error occurs.
+"""
+
+RAISE = 2
+"""If one of the validation modes is set to this value, an exception is raised
+if a value validation error occurs.
+"""
+
+
+class Settings:
+    """Collection of several configuration values.
+    Accessed via the singleton :attr:`settings`.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    reading_validation_mode : int
+        Defines behavior for value validation while reading value.
+        Value validation checks if a value is allowed by the DICOM Standard,
+        e.g. that DS strings are not longer than 16 characters and contain only
+        allowed characters.
+        The default (:attr:`WARN`) is to log a warning in the case of
+        an invalid value, :attr:`RAISE` will raise an error in this
+        case, and :attr:`NO_CHECK` will bypass the
+        validation (with the exception of some encoding errors).
+    writing_validation_mode : int
+        Defines behavior for value validation while writing a value.
+        See :attr:`Settings.reading_validation_mode`.
+    """
+
+    def __init__(self) -> None:
+        self._reading_validation_mode: Optional[int] = None
+        self.writing_validation_mode = RAISE
+
```

Comment:
I know you commented that writing validation would be changed in this PR.  And here  `enforce_valid_values` that it replaces for writing was default `False`.  I like this, but perhaps as a breaking change in pydicom 3.x?  Perhaps it could be put it in using the `config.future_behavior` mechanism for now?

Reply:
I have been thinking about that. My reasoning was that writing DICOM data is done by the user (as opposed to reading data created by someone else), so it would be helpful to do that. But you are right, this would be a breaking change - I will change it and add it to the future mechanism as proposed.

Reply:
And I can also change it similar to `reading_validation_mode` to use ` enforce_valid_values` as fallback.

Reply:
I noticed that the other constants use string values (`"WARN"`, `"RAISE"` and `"IGNORE"`), maybe I should change that too, for consistence?

Reply:
Well, this is new so I think we could do what we want - I don't mind an enum rather than a string, but I like IGNORE a little better than NO_CHECK, then all three are a verb.   But it doesn't really matter, I'm happy with any of these variations.

Reply:
Ok, will change to IGNORE, that makes sense.

Reply:
Done.